### DISCUSSION TITLE
Where is the specification of list_pattern, length_pattern, slice_pattern ?

### DISCUSSION THREAD

**Discussion by gafter** [dotnet/csharplang#5071] [Language Ideas]
I see three new syntactic nonterminals introduced in https://github.com/dotnet/csharplang/blob/main/proposals/list-patterns-enumerables.md but I don't see them defined.

@jcouv

**Comment by 333fred** [comment]
They are defined in https://github.com/dotnet/csharplang/blob/main/proposals/list-patterns.md, though the enumerables  version needs to be updated to remove the length pattern variant as ldm has removed it.

**Reply by gafter** [reply]
Can you please identify what line number `length_pattern` appears at?  I can't find it.

Should https://github.com/dotnet/csharplang/blob/main/proposals/list-patterns-enumerables.md contain a link or reference to that document?

**Reply by jcouv** [reply]
There's not length-patterns anymore, but the "list-patterns on enumerables" spec wasn't yet updated to reflect that (we need to do plain list-patterns first).

**Reply by jcouv** [reply]
Btw, if you'd like to take a look, we'd love your input/feedback: https://github.com/dotnet/roslyn/pull/53891
